pleskdistup/actions/distupgrade.py

```diff
@@ -145,6 +146,140 @@ def estimate_revert_time(self) -> int:
         return 20
 
 
+class CheckAptReposBackups(action.CheckAction):
+    description: str
+    _name: str
+    sources_list_path: str
+    sources_list_d_path: str
+
+    @staticmethod
+    def get_all_repo_list_files(sources_list_path: str, sources_list_d_path: str) -> typing.List[str]:
+        ret = [sources_list_path]
+        for root, _, filenames in os.walk(sources_list_d_path):
+            for f in filenames:
+                if f.endswith(".list"):
+                    ret.append(os.path.join(root, f))
+        return ret
+
+    def __init__(
+        self,
+        sources_list_path: str = "/etc/apt/sources.list",
+        sources_list_d_path: str = "/etc/apt/sources.list.d/",
+    ) -> None:
+        self._name = "Check repo.list leftovers from previous conversion"
+        self.description = "These repo.list files are found with backups, please check/restore them, and delete the backups: {}"
```

Comment:
I think you could ask to use `--revert` here. It seems much simpler than manually searching for files.
In fact, it could be a great way to ensure the customer reverts after a failed conversion, thank you for the idea. :thinking:

Reply:
Everything else looks great, so let's change the description and merge this =)

Reply:
I wasn't sure if those archive files are purely safe so I asked the customer to check.
Asking about a command line switch in a library seems counter-intuitive to me but OK